File: app/src/main/java/app/bolling/chucknorris/dagger/AppComponent.java
package app.bolling.chucknorris.dagger;

import javax.inject.Singleton;

import app.bolling.chucknorris.DataRepository;
import app.bolling.chucknorris.ui.fragment.favourite.FavouriteViewModel;
import app.bolling.chucknorris.ui.fragment.favourite.FavouritesFragment;
import app.bolling.chucknorris.ui.fragment.joke.JokeFragment;
import app.bolling.chucknorris.ui.fragment.joke.JokeViewModel;
import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, RoomModule.class})
public interface AppComponent {
    void inject(DataRepository repository);
    void inject(JokeFragment fragment);
    void inject(JokeViewModel viewModel);
    void inject(FavouritesFragment fragment);
    void inject(FavouriteViewModel viewModel);
}

File: app/src/main/java/app/bolling/chucknorris/dagger/AppComponent.kt
package app.bolling.chucknorris.dagger

import app.bolling.chucknorris.util.DataRepository
import app.bolling.chucknorris.ui.fragment.favourite.FavouriteViewModel
import app.bolling.chucknorris.ui.fragment.favourite.FavouritesFragment
import app.bolling.chucknorris.ui.fragment.joke.JokeFragment
import app.bolling.chucknorris.ui.fragment.joke.JokeViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, RoomModule::class))
interface AppComponent {
    fun inject(repository: DataRepository)
    fun inject(fragment: JokeFragment)
    fun inject(viewModel: JokeViewModel)
    fun inject(fragment: FavouritesFragment)
    fun inject(viewModel: FavouriteViewModel)
}